torch/_decomp/decompositions.py

```diff
@@ -1775,17 +1775,23 @@ def log_sigmoid_forward(self: Tensor) -> Tuple[Tensor, Tensor]:
 
 @register_decomposition(aten.norm)
 @out_wrapper()
-@reduction_complex_to_real
 def norm(
     self: Tensor,
     p: Optional[float] = None,
     dim: List[int] = None,
     keepdim: bool = False,
     dtype: Optional[torch.dtype] = None,
 ):
-    if p is None:
-        p = 2.0
-    return torch.linalg.vector_norm(self, p, dim, keepdim, dtype=dtype)
+    p = p if p is not None else 2.0
+    if dtype:
+        return torch.linalg.vector_norm(self, p, dim, keepdim, dtype=dtype)
+
+    computation_dtype, result_dtype = utils.elementwise_dtypes(
```

Comment:
do you need this? Just calling `vector_norm` should be enough. Alternatively, we can switch to `norm` decomposition from `_refs` (and it's just calling `torch.linalg.vector_norm`, passing through dtype arg).

Reply:
i think it's still good to model the dtype conversion so inductor can fuse it

Reply:
Inductor gets fusable dtype conversions from `linalg_vector_norm` decomposition that is uses, so I really think there's no point in this decomp, we should just add registration to the one in `_refs` that does the same thing.